docs/ContainerRuntimeConfigDesign.md

```diff
@@ -0,0 +1,95 @@
+# Summary
+
+Users need a way to update the container runtime configuration. Users can do this via MCO currently, but they need to know the correct options to use. We can hopefully help customers by documenting a CRD with the ContainerRuntime tuneables and have the MCO use these tuneables when rendering the crio.conf, storage.conf, and registries.conf files to Ignition/disk.
+
+The CRI-O runtime is used by default and is the recommended solution.
+
+# Motivation
+
+The MCO contains the logic to write the crio.conf, storage.conf, and registries.conf configuration files to Ignition. When Ignition starts on a machine it writes these three files to configure the container runtime. When these resources change within the MachineConfig, the new configs are rewritten and the MachineConfigDaemon is instructed to reboot the machine to pick up the new configs.
```

Comment:
Minor: well, the MCD is not explicitly "instructed" (I think this is what #360 is suggesting). The MCD could in the future become smarter and not reboot for things it knows don't require a reboot.